packages/babili/src/cli.js

```diff
@@ -0,0 +1,169 @@
+const yargsParser = require("yargs-parser");
+const optionsParser = require("./options-parser");
+const { processFiles } = require("./fs");
+
+const plugins = [
+  "booleans",
+  "builtIns",
+  "consecutiveAdds",
+  "deadcode",
+  "evaluate",
+  "flipComparisons",
+  "guards",
+  "infinity",
+  "mangle",
+  "memberExpressions",
+  "mergeVars",
+  "numericLiterals",
+  "propertyLiterals",
+  "regexpConstructors",
+  "removeConsole",
+  "removeDebugger",
+  "removeUndefined",
+  "replace",
+  "simplify",
+  "simplifyComparisons",
+  "typeConstructors",
+  "undefinedToVoid"
+];
+
+const proxies = ["keepFnName", "keepClassName"];
+
+const dceBooleanOpts = [
+  "deadcode.keepFnName",
+  "deadcode.keepFnArgs",
+  "deadcode.keepClassName"
+];
+
+const mangleBooleanOpts = [
+  "mangle.eval",
+  "mangle.keepFnName",
+  "mangle.topLevel",
+  "mangle.keepClassName"
+];
+
+const mangleArrayOpts = ["mangle.blacklist"];
+
+const typeConsOpts = [
+  "typeConstructors.array",
+  "typeConstructors.boolean",
+  "typeConstructors.number",
+  "typeConstructors.object",
+  "typeConstructors.string"
+];
+
+const cliBooleanOpts = ["stdin", "help", "version"];
+const cliOpts = ["out-file", "out-dir"];
+const alias = {
+  outFile: "o",
+  outDir: "d"
+};
+
+function aliasArr(obj) {
+  const r = Object.keys(obj).reduce((acc, val) => {
+    return acc.concat(val, obj[val]);
+  }, []);
+  return r;
+}
+
+function validate(opts) {
+  const allOpts = [
+    ...plugins,
+    ...proxies,
+    ...dceBooleanOpts,
+    ...mangleBooleanOpts,
+    ...typeConsOpts,
+    ...mangleArrayOpts,
+    ...cliBooleanOpts,
+    ...cliOpts,
+    ...aliasArr(alias)
+  ];
+
+  return Object.keys(opts).filter(
+    opt => opt !== "_" && allOpts.indexOf(opt) === -1
+  );
+}
+
+function run(args) {
+  const presetOpts = [...plugins, ...proxies];
+
+  const booleanOpts = [
+    ...presetOpts,
+    ...dceBooleanOpts,
+    ...mangleBooleanOpts,
+    ...typeConsOpts,
+    ...cliBooleanOpts
+  ];
+
+  const booleanDefaults = booleanOpts.reduce(
+    (acc, cur) =>
+      Object.assign(acc, {
+        [cur]: void 0
+      }),
+    {}
+  );
+
+  const arrayOpts = [...mangleArrayOpts];
+
+  const arrayDefaults = arrayOpts.reduce(
+    (acc, cur) =>
+      Object.assign(acc, {
+        [cur]: []
+      }),
+    {}
+  );
+
+  const argv = yargsParser(args, {
+    boolean: booleanOpts,
+    array: mangleArrayOpts,
+    default: Object.assign({}, arrayDefaults, booleanDefaults),
+    alias,
+    configuration: {
+      "dot-notation": false
+    }
+  });
+
+  const files = argv["_"];
```

Comment:
Yeah and I did also check ignore the file if it isn't present https://github.com/babel/babili/blob/2b1b16ac05596e65ec77c56a1e3e1b7882991341/packages/babili/src/fs.js#L29